### PR TITLE
[Core] New React Panel component

### PR DESCRIPTION
**Pull request by alisterdev**
A react component that renders a collapsible Bootstrap panel (i.e the one used everywhere in Loris)
image

>Hopefully this will help to standardize these panels across Loris and get rid of `filterControl.js` and such

**Usage:**
```js
<Panel id="panel-id" title="Selection Filter">
 // Panel content goes here
</Panel>
```

### CODE REVIEW

File: jsx/Panel.js [hunk 1]
```diff
@@ -61,5 +62,6 @@ class Panel extends React.Component {
 Panel.propTypes = {};
```
Comment (jacobpenny) [inline jsx/Panel.js]: Let's add the `propTypes` for this component

File: jsx/Panel.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    if (this.state.collapsed) {
+      glyphClass = "glyphicon pull-right glyphicon-chevron-down";
+    }
+
+    return (
+      <div className="panel panel-primary">
+        <div className="panel-heading"
+             onClick={this.toggleCollapsed}
+             data-toggle="collapse"
+             data-target={'#' + this.props.id}
```
Comment (jacobpenny) [inline jsx/Panel.js]: What are `data-toggle` and `data-target` used for?

Comment (alisterdev) [inline jsx/Panel.js]: @jacobpenny Bootstrap uses them to do the smooth collapsing

File: jsx/Panel.js (hunk 1 above)
Comment (alisterdev) [inline jsx/Panel.js]: @jacobpenny Done

**Pull request merged**